python-avd/pyavd/_eos_designs/structured_config/network_services/router_adaptive_virtual_topology.py

```diff
@@ -19,123 +20,95 @@ class RouterAdaptiveVirtualTopologyMixin(Protocol):
     Class should only be used as Mixin to a AvdStructuredConfig class.
     """
 
-    @cached_property
-    def router_adaptive_virtual_topology(self: AvdStructuredConfigNetworkServicesProtocol) -> dict | None:
-        """Return structured config for profiles, policies and VRFs for router adaptive-virtual-topology (AVT)."""
+    @structured_config_contributor
+    def router_adaptive_virtual_topology(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
+        """Set the structured config for profiles, policies and VRFs for router adaptive-virtual-topology (AVT)."""
         if not self.shared_utils.is_cv_pathfinder_router:
-            return None
+            return
 
-        router_adaptive_virtual_topology = {
-            "profiles": self._cv_pathfinder_profiles(),
-            "vrfs": self._cv_pathfinder_wan_vrfs(),
-            "policies": self._cv_pathfinder_policies(),
-        }
+        self._set_cv_pathfinder_profiles()
+        self._set_cv_pathfinder_wan_vrfs()
+        self._set_cv_pathfinder_policies()
 
-        return strip_empties_from_dict(router_adaptive_virtual_topology)
-
-    def _cv_pathfinder_wan_vrfs(self: AvdStructuredConfigNetworkServicesProtocol) -> list:
-        """Return a list of WAN VRFs based on filtered tenants and the AVT."""
+    def _set_cv_pathfinder_wan_vrfs(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
+        """Set the WAN VRFs based on filtered tenants and the AVT."""
         # For CV Pathfinder, it is required to go through all the AVT profiles in the policy to assign an ID.
-        wan_vrfs = []
 
         for vrf in self._filtered_wan_vrfs:
-            wan_vrf = {"name": vrf.name, "policy": f"{vrf.policy}-WITH-CP" if vrf.name == "default" else vrf.policy, "profiles": []}
+            wan_vrf = EosCliConfigGen.RouterAdaptiveVirtualTopology.VrfsItem(
+                name=vrf.name, policy=f"{vrf.policy}-WITH-CP" if vrf.name == "default" else vrf.policy
+            )
 
             # Need to allocate an ID for each profile in the policy, for now picked up from the input.
             policy = get_item(
                 self._filtered_wan_policies,
```

Comment:
Are these filtered wan policies only used here (and maybe in other refactored areas)? If so we should refactor it to return proper classes instead of dicts, so we don't have to use get_item etc.

Reply:
Normally the filtered_* are just returning the same type as the input, but in a filtered AvdIndexedList, which makes it much easier to parse below.

Reply:
will be addressed in wan_utils refactor